Port Scheduler library to Arduino Zero.

PeterVH opened the pull request:
The library from the Due does not work as is on the Zero because it loads and stores the processor state using ldmia/stmia. On cortex m0, these instructions only work with the 'low' registers (r0-r7).

I made an implementation that loads the registers in the low registers first, and then moves them into the high regs.
Tested with MultipleBlink sketch.
Some feedback review on the code is desirable.
See [forum](http://forum.arduino.cc/index.php?topic=352408.0)

cmaglie commented on libraries/Scheduler/src/Scheduler.cpp:
```diff
@@ -64,11 +83,39 @@ static void __attribute__((naked)) __attribute__((noinline)) coopTaskStart(void)
 	asm (
 		"mov   r12, sp;"
-		"stmia r0, {r4-r12, lr};"
+		/* store low registers */
```
Shall this be put inside ifdef too, so we can use the more efficient store instruction when available (without copying the high registers into the low registers)?

cmaglie commented on libraries/Scheduler/library.properties:
```diff
@@ -1,9 +1,9 @@
+paragraph=The Scheduler library enables the Arduino to run multiple functions at the same time. This allows tasks to happen without interrupting each other.</br>This is a cooperative scheduler in that the CPU switches from one task to another. The library includes methods for passing control between tasks.
 category=Other
 url=http://www.arduino.cc/en/Reference/Scheduler
 architectures=sam
```
You should append `,samd` to the `architectures` field.

PeterVH commented:
Yes indeed, that chunk of m0 specific code escaped my attention.
I fixed both issues you brought up and squashed everything in one commit again.

B.t.w. I had big trouble testing this on my Due. Is there something seriously wrong with sam package 1.6.4? I could upload sketches to the Due (prog port) but could not run any simple led blinking program. Thought for a moment my Due was toasted. I reverted to 1.6.3 and everything was back ok.

Anyway test passed on Due too.

q2dg commented:
Should https://www.arduino.cc/en/Reference/Libraries list be updated to reflect Scheduler is "Due and Zero only", then?

If it is, Zero board should be referenced too in first sentence of https://www.arduino.cc/en/Reference/Scheduler ...
